### PR TITLE
Upgrade CI to GitHub workflow.

### PR DESCRIPTION
**Pull request by shulard**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| Documentation   | 
| License         | MIT

#### What's in this PR?

The CI system is migrated from Travis to GitHub Actions. The workflow was taken from `php-http/message` repository which seems to have the more complete configuration.

#### Why?

Since travis is not reliable anymore for OSS projects, we must move to the new GitHub feature.

#### Checklist

- [ ] Updated CHANGELOG.md to describe BC breaks / deprecations | new feature | bugfix
- [ ] Documentation pull request created (if not simply a bugfix)

#### To Do

- [ ] Is Scrutinizer still used somewhere ?

### CODE REVIEW

**Comment by shulard** [comment]
Maybe this change require some tweaks in the project configuration…

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 [![Latest Version](https://img.shields.io/github/release/php-http/react-adapter.svg?style=flat-square)](https://github.com/php-http/react-adapter/releases)
 [![Software License](https://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat-square)](LICENSE)
-[![Build Status](https://img.shields.io/travis/php-http/react-adapter.svg?style=flat-square)](https://travis-ci.org/php-http/react-adapter)
+[![Build Status](https://img.shields.io/github/workflow/status/php-http/react-adapter.svg?style=flat-square)](https://github.com/php-http/react-adapter/actions?query=workflow%3ATests)
```
Comment (dbu) [inline README.md:5]: afaik the shield url needs to contain the name of the workflow you want to show as well: https://img.shields.io/github/workflow/status/php-http/react-adapter/CI.svg?style=flat-square

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
         }
     ],
     "require": {
-        "php": "^7.1",
+        "php": ">=7.1",
```
Comment (dbu) [inline composer.json:14]: i am aware this is controversal. i am in the camp of people who prefer a merge request and release to allow a new php major version over not knowing if a library actually works with that new major version. and composer would then install the correct version if we need to do fixes for php 8 (or 9, sometimes in the future).

i'd prefer `^7.1|^8.0`

Comment (shulard) [reply composer.json:14]: I'm ok with the `^7.1|^8.0` :smile:

**Comment by dbu** [comment]
we seem to use an antique phpunit in this repository. can you try with ^8.5 instead of the version 4 and 5? (and i think we see an example of `"php": ">=5.4"` or something here... phpunit 7 would correctly report that it does not work on php 8. (though phpunit 9 seems to have dropped this again and goes with >=)

**Comment by shulard** [comment]
Yes you're right about phpunit. But I think we must work on merging the #42 first instead of this one. In the #42, I removed the direct dependency to PHPUnit and let the `php-http/client-integration-tests` package define the correct version to use. This adapter package doesn't have any direct interaction with PHPUnit :smile:.

So I'm pretty sure that after merging #42, this phpunit version problem will disappear !

**Comment by shulard** [comment]
Hello !

I just rebased this branch on master and it seems that the workflow is now passing: https://github.com/shulard/react-adapter/runs/1570693771

**Review by dbu — APPROVED** [review]
looks great to me!

**Pull request merged**